processing/src/main/java/org/apache/druid/query/scan/ScanQuery.java

```diff
@@ -153,7 +153,7 @@ public ScanQuery(
     if (order != Order.NONE) {
       Preconditions.checkArgument(
           (columns.contains(ColumnHolder.TIME_COLUMN_NAME)),
```

Comment:
I believe `columns` scan be null or empty, and in this case, all columns would be included. The check here would be invalid then. Could you verify through a unit test, and fix if needed?